### DISCUSSION TITLE
Deploy hugo with staging URL

### DISCUSSION THREAD

**Discussion by stefanvanburen** [vercel/vercel#3549] [Ideas]
Hi,

Hugo uses a baseUrl in its configuration file to figure out links and such during the build process. Is there a way to inject the staging domain URL so that relative links work during staging, while still being able to alias to the production URL once merged to master?

For example, hugo itself has an example of how this works with netlify:
https://github.com/gohugoio/hugoDocs/blob/master/netlify.toml

I'm guessing that using some of the builder's environment vars would enable this, but thought I'd ask if there's a more public example of how this might work.

Thanks!

**Comment by clasense4** [comment]
Hi,
I came across this requirement also. So, I created a simple bash script and set several environment variables. The bash script looks like this.

```bash
#!/usr/bin/env bash

if [ -z ${BASE_URL} ]; then
    echo "BASE_URL is unset, use VERCEL: '$VERCEL_URL'";
    hugo -b https://$VERCEL_URL
else
    echo "BASE_URL is set to '$BASE_URL'";
    hugo -b https://$BASE_URL
fi
```
For environment variables, I set `BASE_URL` for production and `VERCEL_URL` for preview.

And this is the build command
```bash
chmod +x build.sh && ./build.sh
```

Regards

**Comment by tristan957** [comment]
@clasense4 I am for some reason not seeing $VERCEL_URL being populated in my builds. Any ideas? Trying to do something similar to you.

Example failure here https://vercel.com/tristan957/tristan-partin-io/32jf5zzi9 where the command was `hugo -b $VERCEL_URL`

**Comment by clasense4** [comment]
@tristan957 You need to manually create the `VERCEL_URL` and it will be populated automatically. Please also check my screenshot below.

![image](https://user-images.githubusercontent.com/1615045/86222957-23a76580-bbb1-11ea-9625-4ea240762dcd.png)

**Comment by tristan957** [comment]
Wow am I blind. I must not have seen that in the docs. Thanks for your help!

**Comment by SeanHood** [comment]
Found this thread but opted for this one liner version: `hugo -b https://${SITE_URL:-$VERCEL_URL}`, used in the "Build Command" config.

`SITE_URL` is only set in Production, every other env picks up `VERCEL_URL`.

`${SOME_VAR:-default}` is a feature of bash. It will set `default` if `$SOME_VAR` is not set or null.

**Reply by tristan957** [reply]
Really good point Sean. Thanks!

**Comment by almmiko** [comment]
`VERCEL_URL` is used as `System Environment Variable`; when you add your custom domain, `VERCEL_URL` will be set automatically. You don't need to create VERCEL_URL by yourself.

I tested it with 

```bash
#!/usr/bin/env bash

if [[ ${VERCEL_ENV} == "production" ]]; then
    echo "Production deployment with VERCEL_URL: '$VERCEL_URL'";
    hugo -b https://$VERCEL_URL --gc
else
    echo "Not production deployment with VERCEL_URL: '$VERCEL_URL'";
    hugo -b https://$VERCEL_URL -D --gc
fi

```
